static/js/components/ShowAvatar.jsx

```diff
@@ -0,0 +1,58 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { createMuiTheme } from '@material-ui/core/styles';
+
+import Avatar from '@material-ui/core/Avatar';
+
+const theme = createMuiTheme({});
+
+const ShowAvatar = ({ size, firstName, lastName, username, gravatarUrl }) => {
+  // use the hash of the username (which is in the gravatarUrl) to
+  // select a unique color for this user
+  function bgcolor() {
+    let hash = gravatarUrl.split("/");
```

Comment:
Why not just use the username that was passed in?

Reply:
`gravatarUrl` is a hash of the `username` (by default, and of the email address if it's given), already calculated on the Python side in the `User` model:
```
    @property
    def gravatar_url(self):
        email = self.contact_email if \
                self.contact_email is not None else self.username

        digest = md5(email.lower().encode('utf-8')).hexdigest()
        # return a 404 status code if not found on gravatar
        return f'https://www.gravatar.com/avatar/{digest}?d=404'
```
I'm avoiding having to recalculate the hash on the client side.